File: app/controllers/preferences_controller.rb
class PreferencesController < UITableViewController

  def prepareForSegue(segue, sender: sender)
    selectedIndexPath = self.tableView.indexPathForSelectedRow

    if segue.identifier == "EditCategorySegue"
      @needsReload = true
      segue.destinationViewController.category = Category.where(:index).eq(selectedIndexPath.row).first
    end
  end

  def done
    self.navigationController.presentingViewController.dismissModalViewControllerAnimated(true)
  end

  def viewWillAppear(animated)
    if @needsReload
      self.tableView.reloadData
      @needsReload = false
    end
  end

  #### table view delegate methods ####

  def tableView(tableView, numberOfRowsInSection: section)
    Category.count
  end

  def tableView(tableView, cellForRowAtIndexPath: indexPath)
    cell = tableView.dequeueReusableCellWithIdentifier("CategoryCell")
    cell ||= UITableViewCell.alloc.initWithStyle(UITableViewCellStyleDefault, reuseIdentifier: "CategoryCell")
    category = Category.where(:index).eq(indexPath.row).first
    cell.nameLabel.text = category.name
    cell.colorMark.color = category.color

    cell
  end

end

File: Forking Pasta/PreferencesController.swift
import UIKit
import KingPastaKit

class PreferencesController: UITableViewController {
    
    private var needsReload = false
    
    override func viewWillAppear(animated: Bool) {
        super.viewWillAppear(animated)
        
        if needsReload {
            tableView.reloadData()
            needsReload = false
        }
    }
    
    override func prepareForSegue(segue: UIStoryboardSegue, sender: AnyObject?) {
        if let identifier = segue.identifier, let selectedIndexPath = tableView.indexPathForSelectedRow {
            if identifier == "EditTimeZoneSegue" {
                needsReload = true
                (segue.destinationViewController as? EditTimeZoneController)?.timeZone = TimeZone.forIndex(selectedIndexPath.row)
            }
        }
    }
    
    func done() {
        (UIApplication.sharedApplication().delegate as? AppDelegate)?.syncToWatch()
        navigationController?.presentingViewController?.dismissViewControllerAnimated(true, completion: nil)
    }
    
    func timeIntervalDidChange(sender: UISegmentedControl) {
        FkP.useFiveMinuteIntervals = sender.selectedSegmentIndex == 1
    }
    
    // MARK: UITableViewDataSource
    override func numberOfSectionsInTableView(tableView: UITableView) -> Int {
        return 2
    }
    
    override func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        if section == 0 {
            return 1
        } else {
            return TimeZone.count
        }
    }
    
    override func tableView(tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
        if section == 0 {
            return nil
        } else {
            return "Categories"
        }
    }
    
    override func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
        if indexPath.section == 0 {
            let cell = tableView.dequeueReusableCellWithIdentifier("TimePickerIntervalCell") as? TimePickerIntervalCell ?? TimePickerIntervalCell(style: .Default, reuseIdentifier: "TimePickerIntervalCell")
            if FkP.useFiveMinuteIntervals == false { cell.timeIntervalControl.selectedSegmentIndex = 0 }
            return cell
        } else {
            let cell = tableView.dequeueReusableCellWithIdentifier("TimeZoneCell") as? SelectTimeZoneCell ?? SelectTimeZoneCell(style: .Default, reuseIdentifier: "TimeZoneCell")
            let timeZone = TimeZone.forIndex(indexPath.row)
            cell.nameLabel.text = timeZone.name
            cell.colorMark.color = timeZone.color
            
            return cell
        }
    }
    
}